tests/console/btrfs_check.pm

```diff
@@ -0,0 +1,45 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: btrfs_check
+# Summary: Test that btrfs is setup properly
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    select_console 'root-console';
+
+    # Prepare the arrays of subvolumes for comparing
+    # TODO: the subvolumes list vary depending on the arch and the version
+    my @kiwi_volumes = qw(@ @/.snapshots @/home @/opt @/root @/srv @/tmp @/usr/local @/var);
+    my @test_volumes = split("\n", script_output("btrfs subvolume list / | cut -d ' ' -f 9"));
+    record_info('test_volumes', "@test_volumes");
+    record_info('kiwi_volumes', "@kiwi_volumes");
+
+    # Compare that the lists of subvolumes match
+    foreach my $kiwi (@kiwi_volumes) {
+        my $found = 0;
+        foreach my $test (@test_volumes) {
```

Comment:
This can be simplified by using the `any` module from `List::Util` I think

```suggestion
        die "Subvolume $kiwi not present" unless (any {$_ eq $kiwi} @test_volumes);
```

And you will need a `use List::Util 'any';` in the beginning.

I think that can simplify this routine a lot.

Reply:
Let's put this to the test!
https://www.youtube.com/watch?v=sByxTCQU8Rc

Reply:
You're goddamn right!!
https://youtu.be/S9RVS8cjNN0